bootenv.go

```diff
@@ -69,10 +75,17 @@ func (e *uBootEnv) checkEnvCanary() error {
 
 func (e *uBootEnv) ReadEnv(names ...string) (BootVars, error) {
 	if err := e.checkEnvCanary(); err != nil {
+		if os.Geteuid() != 0 {
+			return nil, permissionDeniedError("fw_printenv")
```

Comment:
Sorry to be pedantic, but we don't really know that permissions are the issue here. It could also be that the binary is not found, for example, and here we're hiding the original error. Here's what I suggest:

```
if os.Geteuid() != 0 {
    return nil, errors.Wrap(err, "requires root privileges")
}
```

That way you preserve both errors, and the user is more likely to figure out what went wrong. Note that I'm avoiding the phrase "Permission denied" and just stating that we require root privileges, because we don't know if that's the issue.

Similar for the rest of the diff.

Reply:
Are we not able to catch the error code from "fw_setenv" invocation? I am pretty sure it returned the correct error code (EACCES) when invoked from the CLI.

Reply:
Tested, it prints the correct error string but returns 1.

```
root@colibri-imx7-emmc:~# /sbin/fw_setenv mirza test                                                                                        
Write error on /dev/mmcblk0boot0: Operation not permitted                                                                                   
Error: can't write fw_env to flash                                                                                                          
root@colibri-imx7-emmc:~# echo $?                                                                                                           
1
```

Reply:
Made the changes @kacf requested.